Code/skyrim_ui/src/app/models/popup-notification.ts

```diff
@@ -1,22 +1,18 @@
-import { Subject } from 'rxjs';
-import { Player } from './player';
-
-
-export enum NotificationType {
-  Connection = 1,
-  Invitation
-}
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
 
 export interface PopupNotification {
-  message: string;
-  type?: NotificationType;
-  player?: Player;
+  /** Translation Key. Plaintext should **NOT** be used */
+  messageKey: string;
+  messageParams?: Record<string, any>;
+  imageUrl?: string;
+  icon?: IconDefinition;
+  /** milliseconds */
   duration?: number;
-  onClose: Subject<boolean>;
+  actions?: PopupNotifactionAction[];
```

Comment:
Back to callback hell I guess 😛

Reply:
I get where you're coming from, but I think callbacks work just fine for such a simple system. Could also go with Promises, but I don't see much benefit unless you need them for chaining